akka-docs/rst/java/stream/stages-overview.rst

```diff
@@ -959,6 +959,89 @@ merging. The maximum number of merged sources has to be specified.
 **completes** when upstream completes and all consumed substreams complete
 
 
+Time aware stages
+-----------------
+
+Those stages operates taking time into consideration.
+
+initialTimeout
+^^^^^^^^^^^^^^
+If the first element has not passed through this stage before the provided timeout, the stream is failed
+with a ``TimeoutException``.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses before first element arrives
+
+**cancels** when downstream cancels
+
+completionTimeout
+^^^^^^^^^^^^^^^^^
+If the completion of the stream does not happen until the provided timeout, the stream is failed
+with a ``TimeoutException``.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses before upstream completes
+
+**cancels** when downstream cancels
+
+idleTimeout
+^^^^^^^^^^^
+If the time between two processed elements exceed the provided timeout, the stream is failed
+with a ``TimeoutException``.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses between two emitted elements
+
+**cancels** when downstream cancels
+
+backpressureTimeout
+^^^^^^^^^^^^^^^^^^^
+If the time between emission of element and following downstream demand exceed the provided timeout,
+the stream is failed with a ``TimeoutException``. If producer doesn't emmit any element,
+the timeout is still valid and will fail the stream.
+
+**emits** when upstream emits an element
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes or fails if timeout elapses between element emission and downstream demand.
+
+**cancels** when downstream cancels
+
+keepAlive
+^^^^^^^^^
+Injects additional (configured) elements if the upstream does not emit for a configured amount of time.
+
+**emits** when upstream emits an element or when configured amount of time has passed
+
+**backpressures** when downstream backpressures
+
+**completes** when upstream completes
+
+**cancels** when downstream cancels
+
+initialDelay
+^^^^^^^^^^^^
+Delays the initial element by the specified duration.
+
+**emits** when upstream emits an element if the initial delay already elapsed
```

Comment:
"**with** the initial delay already elapsed" or "if the initial delay **is** already elapsed"?